imageatm/components/cloud.py

```diff
@@ -73,37 +78,33 @@ def _set_remote_dirs(self):
         cmd = '{} mkdir -p {}'.format(self.ssh, self.remote_workdir)
         run_cmd(cmd, logger=self.logger)
 
-        self.remote_image_dir = os.path.join(self.remote_workdir, os.path.basename(self.image_dir))
-        self.remote_job_dir = os.path.join(self.remote_workdir, os.path.basename(self.job_dir))
+        self.remote_image_dir = self.remote_workdir.joinpath(self.image_dir.name)
+        self.remote_job_dir = self.remote_workdir.joinpath(self.job_dir.name)
 
     def _set_s3_dirs(self):
-        if 's3://' in self.image_dir:
+        if isinstance(self.image_dir, URL):
             self.s3_image_dir = self.image_dir
         else:
-            self.s3_image_dir = os.path.join(
-                self.s3_bucket, 'image_dirs', os.path.basename(self.image_dir)
-            )
+            self.s3_image_dir = self.s3_bucket / 'image_dirs' / self.image_dir.name
 
-        if 's3://' in self.job_dir:
+        if isinstance(self.job_dir, URL):
             self.s3_job_dir = self.job_dir
         else:
-            self.s3_job_dir = os.path.join(
-                self.s3_bucket, 'job_dirs', os.path.basename(self.job_dir)
-            )
+            self.s3_job_dir = self.s3_bucket / 'job_dirs' / self.job_dir.name
 
     def _sync_local_s3(self):
         self.logger.info('Syncing files local <> s3...')
         self._set_s3_dirs()
 
         # only sync if image dir is local dir
-        if 's3://' not in self.image_dir:
+        if isinstance(self.image_dir, Path):
```

Comment:
Previous code is more specific in my opinion. 
```suggestion
        if 's3://' not in str(self.image_dir):
```